File: backend/routers/code2Token.py
# 경로->auth/github

from pydantic import BaseModel
import requests ,os
from dotenv import load_dotenv
from fastapi import APIRouter

router=APIRouter() #객체생성

class GithubCodeRequest(BaseModel):
    code:str

load_dotenv()    
client_id=os.getenv("GITHUB_CLIENT_ID")
client_secret=os.getenv("GITHUB_CLIENT_SECRET")
exchange_token_url="https://github.com/login/oauth/access_token"

@router.post("/get/token")
def github_auth(request:GithubCodeRequest):

    token_params={
        'client_id':client_id,
        'client_secret':client_secret,
        'code':request.code,
        'redirect_uri':'exp://localhost:8081/--/oauth/callback'
    }
    headers={
        'Accept':'application/json'
    }
    response=requests.post(
        exchange_token_url,
        data=token_params,
        headers=headers
    )
    res=response.json()
    access_token=res['access_token']
    print('access_token: ',access_token)

    return {"success":True,"access_token":access_token}

File: api/code2token.js
const getToken = async (code)=>{
    try{
        const response=await fetch('http://localhost:8082/get/token',{
            method:'POST',
            headers:{'Content-Type':'application/json'},
            body:JSON.stringify({code})
        })
        const result=await response.json()
        return result
        //console.log('토큰: ',result.access_token)//형변환이 아님, 데이터 받아오고 있기때문에 await
        //await SecureStore.setItemAsync("GITHUB_ACCESS_TOKEN",result.access_token) //key:value 형태로 저장. getItemAsync(key)로 가져올수있음
        //router.replace('/home')
    }
    catch (error){
        console.log('error: ',error ,'-',"access token 변환 실패")
    }
}
export default getToken